feat(highlights): add mason support

antoineco opened the pull request:
Improves the highlighting of [Mason](https://github.com/williamboman/mason.nvim) windows (a popular installer for external tools such as LSP servers), which is inconsistent with the colorscheme by default.

_before_

![mason_before](https://github.com/ramojus/mellifluous.nvim/assets/3299086/8bb0efb4-3b5b-42a1-abc3-66d5e2dfeec4)
![mason_g_before](https://github.com/ramojus/mellifluous.nvim/assets/3299086/e794a5ca-e1a4-4744-949b-a6efe7c42535)

_after_

<details><summary>Outdated screenshots</summary>
<p>

![mason_after](https://github.com/ramojus/mellifluous.nvim/assets/3299086/1d439088-44a1-4faa-8c1e-9b67e22b380c)
![mason_g_after](https://github.com/ramojus/mellifluous.nvim/assets/3299086/48263584-830f-4574-a84f-79b236d9c56c)

</p>
</details>

windowsrefund commented:
Looks great. Looking forward to seeing this merged. By the way, no need to update doc/mellifluous.txt as I learned it is automatically updated.

antoineco commented:
@windowsrefund thanks for the hint!

windowsrefund commented:
@antoineco Thank you for all these PRs. Exciting!

ramojus commented:
Everything is great otherwise, thanks! Sorry for the delay on this one, Mason UI was not working in my setup for some reason.

antoineco commented:
@ramojus No problem!

> Everything is great otherwise

The "otherwise" makes me guess that you had a change in mind, but possibly didn't submit the review comment?

antoineco commented on lua/mellifluous/highlights/plugins/mason.lua:
```diff
@@ -0,0 +1,15 @@
+function M.set(hl, colors)
+    hl.set("MasonHeader", { fg = colors.dark_bg, bg = colors.ui_orange, bold = true })
+    hl.set("MasonHeaderSecondary", { fg = colors.dark_bg, bg = colors.ui_blue, bold = true })
+    hl.set("MasonHighlightBlockBold", { fg = colors.dark_bg, bg = colors.ui_blue, bold = true })
```
Got it. I'll address this today.

`Secondary` elements have a different color from the regular (primary) ones in Mason's default config, but in your example Secondary is a link to primary. Do I understand correctly that you want both primary and Secondary menus to look the same in mellifluous?

ramojus replied:
Yes, I don't think it needs to be different color. The only time I've observed the `MasonHighlightBlockBoldSecondary` highlight being shown is when help is toggled. In that case, `MasonHeader` also changes to `MasonHeaderSecondary` and the text clearly indicates that help is toggled, so I think it's clear that a different thing is being shown.

antoineco commented:
Here is an updated screenshot after addressing the suggestions from the review: <img width="814" alt="Screenshot 2024-05-31 at 15 37 29" src="https://github.com/ramojus/mellifluous.nvim/assets/3299086/e06111df-3840-4eb5-bd9e-3f284d3b60f1">
